### DISCUSSION TITLE
Simple Recipe Import from word, pdf, etc

### DISCUSSION THREAD

**Discussion by coolkau** [mealie-recipes/mealie#2812] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

From earlier times, family, etc, I have a lot reciepes in one-note, word, pdf, etc... 
My main blocker to use mealie more in the family is, that there is no easy way to bulk upload or even quickly upload them into mealie. Currently you need to manually add every new ingredient in a separate field. I would hope that I can at least copy paste a bullet list into mealie and it rolls out the bullets into ingredients...

As a consequence, less than 10% of  my frequently used recipes are uploaded. It would be great if there is a way to upload them into mealie. 

### Please provide a concise description of the feature that would resolve your issue.

Suggested options:

- Ability to copy and paste plain text into mealie; the scraper will assign the relevant text snippets to the right sections, roll out the ingredients etc

It is easy enough to transform the recipes into any format outside of mealie. 
- So as a minimum option, uploading recipe jsons would work. This might as simple as a simplied version of the current recipe-download-json. 

### Please consider and list out some caveats or tradeoffs made in your design decision

- Option 1 is more work,  
- Option 2 is really only a tool for people with coding skill. But from a workload perspective it only be a json parser. 

### Additional Information

- [X] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by monksy** [comment]
I think this feature request could be broken up into "Parse from text" and bulk upload from text based PDF or word doc.

**Comment by coolkau** [comment]
With the raise of ChatGPT I am even more keen on having a very simple version of this integrated in mealie. I can use an LLM to bring my recipes into the json format required for uploading. So all we need really is a json parser which checks in the incoming recipes. This should be more or less ready from the backup ingestion, but it should most likely more forgiving with missing fields. Then we only need a mention on the doc page describing the json format. 

And all I need to do is creative prompt writing to get them all in shape

**Reply by Kuchenpirat** [reply]
You would actually not have to change anything regarding mealie, because you can simply use the api to create your recipes.

`/docs` has an description of all available endpoints and also allows you to see the required data.
